common/src/gas_provider/node.rs

```diff
@@ -39,6 +65,12 @@ pub enum GasNode<ExternalId: Clone, Id: Clone, Balance: Zero + Clone> {
     /// (not node's parent, not node's child).
     ReservedLocal { id: ExternalId, value: Balance },
 
+    /// A node used for gas reservation feature.
+    ///
+    /// Such node types are independent and aren't part of the tree structure
+    /// (not node's parent, not node's child).
+    Reserved { id: ExternalId, value: Balance },
```

Comment:
I think it's time to re-think names for `ReservedLocal` and `Reserved` nodes in accordance in accordance to their purposes. At least one of them must be changed, because they all have similar: names, instantiation logic, place in the gas tree (forest).